### PR TITLE
Update OCP version for sail-operator e2e test jobs

### PR DESCRIPTION
**Pull request by fjglira**
According to OCP release life cycle, OCP 4.16 is GA, and 4.15 will end full support in September of this year. I'm updating the job to cover the most recent GA version and the next version (4.17): https://access.redhat.com/support/policy/updates/openshift

### CODE REVIEW

**Comment by fjglira** [comment]
/pj-rehearse periodic-ci-openshift-service-mesh-sail-operator-main-nightly-images pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-16 pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17

**Comment by fjglira** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17 pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-16

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml [hunk 1]
```diff
@@ -87,12 +92,12 @@ tests:
...
     name: ossm-github-simple-job
   timeout: 20m0s
-- as: e2e-ocp-4-15
+- as: e2e-ocp-4-16
   steps:
     cluster_profile: ossm-aws
     env:
       BASE_DOMAIN: servicemesh.devcluster.openshift.com
-      HYPERSHIFT_HC_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.15.6-multi
+      HYPERSHIFT_HC_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.16.3-multi
```
Comment (FilipB) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:100]: How does this work? Do we have to always change this to get newer OCP version?

**Review by FilipB — APPROVED** [review]

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml (hunk 1 above)
Comment (fjglira) [reply ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:100]: We are the only ones using this variable: HYPERSHIFT_HC_RELEASE_IMAGE. It seems to be that is the way to update but is manual work. On ipi cluster we can set the version and the channel

**Pull request merged**